python/lsst/rubintv/production/slac/mosaicing.py

```diff
@@ -39,8 +40,19 @@
 
 from ..utils import isFileWorldWritable
 
+if TYPE_CHECKING:
+    from logging import Logger
 
-def getBinnedFilename(expId, instrument, detectorName, dataPath, binSize):
+    from matplotlib.pyplot import Figure
+
+    from lsst.afw.cameraGeom import Camera, Detector
+    from lsst.afw.display import Display
+    from lsst.afw.image import Exposure, Image
+    from lsst.daf.butler import Butler, DeferredDatasetRef
+    from lsst.pipe.base import Struct
```

Comment:
You've got the parent modules of a lot of these imported already.  You should really stick to one pattern or another, at least within a single file.  It will be very easy given what you've got here to e.g. `return Struct(...)` in some new code and not notice until runtime that it needed to be `return pipeBase.Struct(...)`.